drivers/crc/crc_hw.h

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2024 Brill Power Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_CRC_H
+#define ZEPHYR_INCLUDE_DRIVERS_CRC_H
+
+#include "autoconf.h"
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @struct crc_data
+ * @brief Runtime driver configuration
+ * @param mutex Zephyr mutex object to make driver thread-safe when driver
+ *				is used within multiple threads
+ */
+struct crc_data {
+#ifdef CONFIG_CRC_THREADSAFE
+	struct k_mutex mutex;
+	k_timeout_t crc_8_timeout;
+	k_timeout_t crc_32_timeout;
+#endif /* CONFIG_CRC_THREADSAFE */
+};
+
+#ifdef CONFIG_CRC_HW
+
+typedef int (*crc_api_setup_crc_8_ccitt)(const struct device *dev);
+typedef uint8_t (*crc_api_crc_8_ccitt)(const struct device *dev, uint8_t initial_value,
+				       const void *buf, size_t len);
+typedef uint8_t (*crc_api_verify_crc_8_ccitt)(const struct device *dev, uint8_t expected,
+					      uint8_t initial_value, const void *buf, size_t len);
+typedef int (*crc_api_setup_crc_32_ieee)(const struct device *dev);
+typedef uint32_t (*crc_api_crc_32_ieee)(const struct device *dev, const void *buf, size_t len);
+typedef uint32_t (*crc_api_verify_crc_32_ieee)(const struct device *dev, uint32_t expected,
+					       const void *buf, size_t len);
+
+/**
+ * @struct crc_driver_api
+ * @brief Publicly available API accessible as part of device struct
+ */
+struct crc_driver_api {
```

Comment:
This structure declaration is lacking the `__subsystem` annotation (from `<zephyr/toolchain.h>`).